test/e2e/util.go

```diff
@@ -446,6 +449,71 @@ func DeleteRC(c *client.Client, ns, name string) error {
 	return nil
 }
 
+// podInfo contains pod information useful for debugging e2e tests.
+type podInfo struct {
+	oldHostname string
+	oldPhase    string
+	hostname    string
+	phase       string
+}
+
+// PodDiff is a map of pod name to podInfos
+type PodDiff map[string]*podInfo
+
+// Print formats and prints the give PodDiff.
+func (p PodDiff) Print(ignorePhases util.StringSet) {
+	for name, info := range p {
+		if ignorePhases.Has(info.phase) {
+			continue
+		}
+		if info.phase == nonExist {
+			Logf("Pod %v was deleted, had phase %v and host %v", name, info.phase, info.hostname)
+			continue
+		}
+		phaseChange, hostChange := false, false
+		msg := fmt.Sprintf("Pod %v ", name)
+		if info.oldPhase != info.phase {
+			phaseChange = true
+			if info.oldPhase == nonExist {
+				msg += fmt.Sprintf("in phase %v ", info.phase)
+			} else {
+				msg += fmt.Sprintf("went from phase: %v -> %v ", info.oldPhase, info.phase)
+			}
+		}
+		if info.oldHostname != info.hostname {
+			hostChange = true
+			if info.oldHostname == nonExist || info.oldHostname == " " || info.oldHostname == "" {
```

Comment:
In theory never, but one can always create and rc with a pod teamplate that has spec.host = " " and it would be pretty pointless to print that out. Anyway I've removed it, as it can lead to confusion.